### PR TITLE
Initial DB and Docker set up

### PR DESCRIPTION
**Pull request by alexmcdoodz**

### CODE REVIEW

File: src/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM python:3.9
```
Comment (aaron1brooker) [inline src/Dockerfile]: change to 3.10

**Comment by alexmcdoodz** [comment]
Whole application can now be run within docker with the command "docker-compose up --build". You can then open the site as usual using local host.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# FROM python:3.10
+
+# RUN pip install mysql-connector-python
+
+# WORKDIR /usr/app/src
+
+# COPY db_interactions.py ./
+
+FROM python:3.9.10-alpine3.14
```
Comment (aaron1brooker) [inline Dockerfile]: 3.10?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+WORKDIR /src
+RUN pip install --upgrade pip
```
Comment (aaron1brooker) [inline Dockerfile:3]: pip install from requirements.txt

File: src/db_interactions.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import mysql.connector
```
Comment (aaron1brooker) [inline src/db_interactions.py]: need to add package requirements.txt

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM python:3.10-alpine3.14
+WORKDIR /src
+RUN pip install --upgrade pip
+COPY . /src
+RUN pip install -r /src/requirements.txt
+ENV FLASK_APP=app
```
Comment (aaron1brooker) [inline Dockerfile:6]: Requirements.txt is not in src?

File: src/db_interactions.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import mysql.connector  # type: ignore
+
+connection = mysql.connector.connect(user="root", password="root", host="mysql", port="3306", database="db")
+print("Succesfully connected to db")
```
Comment (aaron1brooker) [inline src/db_interactions.py:4]: Can we setup logger?

File: src/db_interactions.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+cursor = connection.cursor()
+cursor.execute("SELECT * FROM items")
+items = cursor.fetchall()
+connection.close()
+
+print("results are: ", items)
```
Comment (aaron1brooker) [inline src/db_interactions.py:11]: Same again

File: Dockerfile (hunk 5 above)
Comment (alexmcdoodz) [reply Dockerfile:6]: im copying the entire root directory into a folder ./ src in the container, so requirements.txt is in /src

File: src/db_interactions.py (hunk 6 above)
Comment (alexmcdoodz) [reply src/db_interactions.py:4]: will set up logger in next pr with actual DB implementations

File: src/db_interactions.py (hunk 7 above)
Comment (alexmcdoodz) [reply src/db_interactions.py:11]: as above comment

**Review by aaron1brooker — APPROVED** [review]

**Pull request merged**